Terraform/EC2/main.tf

```diff
@@ -0,0 +1,377 @@
+#EC2 Bucket Creation
+
+terraform {
+   backend "s3" {
+   region = "us-west-2"
+   bucket = "project1-terraform-backend-bucket"
+   key    = "EC2/terraform.tfstate"
+   dynamodb_table = "project1-terraform-backend-table"
+  }
+}
+
+provider "aws" {
+  version = "~> 2.33"
+  region = "us-west-2"
+  shared_credentials_file = "$HOME/.aws/credentials"
+  profile = "default"
+}
+
+/*
+#IAM_Role
+*/
+resource "aws_iam_role" "instance_connect" {
+  name        = "instance-connect"
+  description = "privileges for the instance-connect demonstration"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": ["ec2.amazonaws.com", "ssm.amazonaws.com" ]
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+/*
+#IAM Role Attachment
+*/
+resource "aws_iam_role_policy_attachment" "instance_connect" {
+  role       = "${aws_iam_role.instance_connect.id}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM"
+}
+/*
+#IAM Instance Profile
+*/
+resource "aws_iam_instance_profile" "instance_connect" {
+  name = "instance-connect"
+  role = "${aws_iam_role.instance_connect.id}"
+}
+/*
+#IAM policy
+*/
+resource "aws_iam_policy" "instance_connect" {
+  name        = "instance-connect"
+  path        = "/test/"
+  description = "Allows use of EC2 instance connect"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+  		"Effect": "Allow",
+  		"Action": "ec2-instance-connect:SendSSHPublicKey",
+  		"Resource": "*",
+  		"Condition": {
+  			"StringEquals": { "ec2:osuser": "ec2-user" }

```

Comment:
These user names I believe are not for Ubuntu O/S for Ubuntu the use name is "ubuntu"

Reply:
changed to ubuntu